docs/new-style-script-objects.md

```diff
@@ -5,7 +5,7 @@ A new-style object allows the user to write a script that executes in an environ
 A new-style script object is defined in a script file by a single object that is assigned to `module.exports`. A new-style script object has the following keys:
 
   * *requires*: An object whose keys define the requirements for executing this script object's functionality.
-  * *function*: A possibly asynchronous function that actually performs the script object's actions. The function will be invoked with no parameters. Any value returned by this function is written into browsertime's result output.
+  * *function*: A possibly asynchronous function that actually performs the script object's actions. The function will be invoked with no parameters. If the function is asynchronous, it must return a Promise. The value `resolve`d by that promise is written into browsertime's result output. If the function is not asynchronous, any value returned by this function is written into browsertime's result output.
```

Comment:
nit: a `Promise`.

Reply:
s/written/serialized/

Also, it's not clear to me what Marionette/Web Driver do for things that can't serialize across the target/host boundary, like functions (and, presumably `Promise` instances).  You don't need to figure it out, but if you know what the algorithm is, tell me.